js/event.js

```diff
@@ -196,26 +197,26 @@ export const addEventHeardAboutStudy = () => {
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox4] = document.getElementById('checkbox4').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox5] = document.getElementById('checkbox5').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox6] = document.getElementById('checkbox6').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox7] = document.getElementById('checkbox7').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox8] = document.getElementById('checkbox8').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox9] = document.getElementById('checkbox9').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox10] = document.getElementById('checkbox10').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox11] = document.getElementById('checkbox11').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox12] = document.getElementById('checkbox12').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox13] = document.getElementById('checkbox13').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox14] = document.getElementById('checkbox14').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox15] = document.getElementById('checkbox15').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox16] = document.getElementById('checkbox16').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox17] = document.getElementById('checkbox17').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox18] = document.getElementById('checkbox18').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox19] = document.getElementById('checkbox19').checked ? fieldMapping.yes : fieldMapping.no;
         
```

Comment:
Above `formData[fieldMapping.heardAboutStudyForm]` is kind of long. We can use a variable to hold the data (eg `let inputData = {};`), and assign the data to `formData` after finishing updates (eg: `formData[fieldMapping.heardAboutStudyForm]= inputData;`)

Reply:
Another idea for shortening things is to store the concepts in the constants file as such:

"heardAboutStudyList": {
checkbox1: 142654897.196856782,
checkbox2: 142654897.461488577,
etc...
}


Also one thing I'm thinking about now, should the names of those constants have a little more meaning than just "checkbox[number]"? I.E. the names could be "public event", or "familyOrFriend". Just a thought, don't feel too strongly one way or the other.

Reply:
Solved